### PR TITLE
Make helm_repository a data resource

### LINKED ISSUE #217: Migrate repository to a data provider
There are several open issues about issues with the `helm_repository` resource. After discussion we came to the conclusion to not have this as a resource as there is no remote state.

The `helm_reposoitory` resource should be re-implemented as a data resource with the old `helm_reposoitory` resource being shim to the new data source to not break backwards compatibility.

### PR DESCRIPTION
**Pull request by meyskens**
Fixes #217

### CODE REVIEW

**Comment by meyskens** [comment]
Hey @alexsomesan my vendor seems to have gotten files added when I ran a `go build` (i tried not to commit them but one got in anyway, oops)

**Comment by alexsomesan** [comment]
@meyskens I noticed this behaviour too. For me it happened on `golang.org/x` stuff. I wasn't able to pinpoint why, but I noticed a couple of vaguely related issues in golang github. Apparently there are improvements on this in Go 1.12. We can try the beta and see if it behaves differently.

I'll keep digging into it and will share what I find.

**Comment by meyskens** [comment]
@alexsomesan I see a lot of .pem files being pulled in, now 1.12 is on rc1 happy to try that one out

**Comment by meyskens** [comment]
I have tested this on our dev terraform setup and messed with my local repo state and this worked fine!

**Comment by rporres** [comment]
Looks terrific, @meyskens. It's only missing documentation

**Comment by meyskens** [comment]
@rporres Thanks! totally forgot!

File: website/docs/repository.html.markdown [hunk 1]
```diff
@@ -54,3 +54,7 @@ The `metadata` block supports:
 ## Import
 
 `helm_repository` does not support import
+
+## Old resource helm_repository
+
+Before 0.9.0 `helm_repository` was a resource and not a data source. The old resource is now a shim to the data source to preserve backwards compatibiliy. As the use of the resource is deprecated it is strongly suggested to move to the new data source.
```
Comment (rporres) [inline website/docs/repository.html.markdown]:
```suggestion
Before 0.9.0 `helm_repository` was a resource and not a data source. The old resource is now a shim to the data source to preserve backwards compatibility. As the use of the resource is deprecated it is strongly suggested to move to the new data source as the compatibility will be removed in a future release.
```

File: website/docs/repository.html.markdown [hunk 2]
```diff
@@ -54,3 +54,7 @@ The `metadata` block supports:
 ## Import
```
Comment (rporres) [inline website/docs/repository.html.markdown]: let's remove this Import section

**Comment by rporres** [comment]
Looks great. I'd just remove the WIP mention of the PR description

The only thing missing here is the vendor issue... cc @alexsomesan

**Comment by rporres** [comment]
No need to check this, @alexsomesan as vendor files have been removed

File: website/docs/repository.html.markdown [hunk 3]
```diff
@@ -6,21 +6,21 @@ description: |-
...
 
 ```hcl
-resource "helm_repository" "incubator" {
+data "helm_repository" "incubator" {
     name = "incubator"
     url  = "https://kubernetes-charts-incubator.storage.googleapis.com"
 }
 
-resource "helm_release" "my_cache" {
+data "helm_release" "my_cache" {
```
Comment (pdecat) [inline website/docs/repository.html.markdown:23]: `helm_release` should have stayed untouched, isn't it?

File: website/docs/repository.html.markdown [hunk 4]
```diff
@@ -6,21 +6,21 @@ description: |-
 
 ---
 
-# Resource: helm_repository
+# Data Source: helm_repository
```
Comment (pdecat) [inline website/docs/repository.html.markdown:9]: In other providers documentation, datasources are usually placed in a `d` folder, separate from resources which are in an `r` folder.

This is mostly important when both a datasource and a resource exist with the same name, e.g. https://www.terraform.io/docs/providers/kubernetes/d/service.html vs https://www.terraform.io/docs/providers/kubernetes/r/service.html

**Pull request merged**